### ISSUE TITLE
Integrate eForm with Survey WebCard

### ISSUE THREAD

**Issue by a6wu** [UCSD/campus-mobile#1011]
ITS-Pro time card: MA-85

Active survey: https://eforms.ucsd.edu/view.php?id=537412

This is complete when:

- [x] The Student Survey Card displays the active survey (Currently https://eforms.ucsd.edu/view.php?id=537412)
- [x] The student can only submit the survey once.
- [x] When the student has completed a survey, display a message "Thank you for completing the survey!"
- [x] The active survey can be changed remotely, without an app update
- [x] [PR 1126 Test Plan](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7BA4EF9569-4B15-4035-8685-92D59F66E35D%7D&file=PR-1126-Test-Plan-7.4.0-QA-1202.xlsx&action=default&mobileredirect=true) PASS on iOS (@gmihir)
- [x] [PR 1126 Test Plan](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7BA4EF9569-4B15-4035-8685-92D59F66E35D%7D&file=PR-1126-Test-Plan-7.4.0-QA-1202.xlsx&action=default&mobileredirect=true) PASS on Android (@asg003)

**Comment by a6wu** [comment]
Update from @sramcode 

Theres a few steps to implement that feature. 
	1.	add this variable to the user profile model (don't forget to update the UserProfileModel.fromJson and UserProfileModel.toJson methods)
	2.	re-generate user_profile_model.g.dart to save the value between launches in hive
	3.	use a similar method to the toggleLot method inside of parking_data_provider.dart (at the end of the method the user-profile is changed and uploaded to the server. line 142-143 in experimental branch)

I completed steps 1 and 2 on Friday, and I began step 3. Today i plan on continuing step three (im super busy with classes on mondays until 7pm but I will try to work on it during my breaks). After i complete step 3, i should be able to store whether or not the user has completed the survey, and prevent the duplicate submission of the survey. Lmk if you have any questions! Thank you.

**Comment by gmihir** [comment]
iOS passes as per https://ucsdcloud-my.sharepoint.com/:x:/g/personal/c3bryant_ucsd_edu/EcA4dw9pkltNpnNE5sKEgdwBvXlqt0u-JG9G49G4uPzZiw?email=migupta%40UCSD.EDU&e=4%3AsKcNZB&at=9.

**Comment by gmihir** [comment]
We ran into some weird results when testing with Charles and Anamika. The survey card showed a message saying "No surveys are available," even though the survey was enabled on the CMS. After digging into this issue, we found the (likely) cause:

The response is received correctly, but it is not parsed correctly by the SurveyModel. 

` response:
I/flutter (11244): 
I/flutter (11244):   {
I/flutter (11244):     "survey_active": true,
I/flutter (11244):     "survey_id": "537412",
I/flutter (11244):     "survey_url": "https://eforms.ucsd.edu/view.php?id=537412"
I/flutter (11244):   }`

`survey model: {survey_active: false, survey_id: 537412, survey_url: https://eforms.ucsd.edu/view.php?id=537412}`

Above, the survey_active field in the response is true. However, it is parsed to be false in the model. This results in the "no surveys available" message being shown.
